api/src/pcapi/core/finance/api.py

```diff
@@ -2141,12 +2141,12 @@ def expire_current_deposit_for_user(user: users_models.User) -> None:
 def _can_be_recredited(user: users_models.User) -> bool:
     return (
         user.age in conf.RECREDIT_TYPE_AGE_MAPPING
-        and _has_celebrated_birthday_since_registration(user)
+        and _has_been_recredited_since_registration(user)
```

Comment:
Je trouve que le nom est troublant ici, on a une condition qui fait `has_been_recredited... and not has_been_recredited`, et d'ailleurs ce n'est pas ce qui est testé.
Je pense qu'on peut changer le nom de la fonction vers `_has_celebrated_birthday_since_credit_or_registration` par exemple.